server/s3_get_object_action.cc

```diff
@@ -760,6 +760,26 @@ void S3GetObjectAction::read_object_data() {
     } else {
       blocks_to_read = total_blocks_to_read - blocks_already_read;
     }
+    // Below is only for first block read op and non-zero block start offset.
+    // For the first block read op, if reading starts from non-zero offset
+    // adjust number of 'blocks_to_read', so that the read does not go
+    // beyond the object size.
+    if (blocks_already_read == 0 && (motr_reader->get_last_index() != 0)) {
```

Comment:
please test with different scenarios
range falling within a part/across parts/within last part/range as complete part

Reply:
Tested scenarios (range falling within part, across parts, within last part) with 1 and 8 motr_units_per_request.